How to build Dockerfile using Alpine Linux

Have a customer trying to build DVT using an Alpine Linux Dockerfile and build fails.

Sample Dockerfile:
```
FROM python:3.12.1-alpine

# Allow statements and log messages to immediately appear in the Knative logs
ENV PYTHONUNBUFFERED True

RUN apk update
# Pre-reqs (gcc for numpy, cmake for pyarrow)
RUN apk add curl build-base gcc cmake unixodbc-dev

# Install DVT
ENV DVT_HOME=/opt/dvt
ENV VIRTUAL_ENV=${DVT_HOME}/.venv
ENV PIPOPTS="--root-user-action=ignore"
RUN mkdir ${DVT_HOME}
RUN python3 -m venv ${VIRTUAL_ENV}
ENV PATH="${VIRTUAL_ENV}/bin:$PATH"
RUN . ${VIRTUAL_ENV}/bin/activate && pip install ${PIPOPTS} --upgrade pip
RUN . ${VIRTUAL_ENV}/bin/activate && pip install ${PIPOPTS} google_pso_data_validator
RUN . ${VIRTUAL_ENV}/bin/activate && pip install ${PIPOPTS} pyodbc

# Entrypoint
ENTRYPOINT ["python","-m","data_validation"]
```

Error comes from Pyarrow:
```
      CMake Error at CMakeLists.txt:268 (find_package):
        By not providing "FindArrow.cmake" in CMAKE_MODULE_PATH this project has
        asked CMake to find a package configuration file provided by "Arrow", but
        CMake did not find one.

        Could not find a package configuration file provided by "Arrow" with any of
        the following names:

          ArrowConfig.cmake
          arrow-config.cmake

        Add the installation prefix of "Arrow" to CMAKE_PREFIX_PATH or set
        "Arrow_DIR" to a directory containing one of the above files.  If "Arrow"
        provides a separate development package or SDK, be sure it has been
        installed.
```

--- a/ci/build.sh
+++ b/ci/build.sh
@@ -14,6 +14,12 @@
 # limitations under the License.
 
 # Copied from https://github.com/googleapis/synthtool/blob/master/synthtool/gcp/templates/python_library/.kokoro/build.sh
+#
+
+PYTHON="python3.9"
+if [[ -n "$1" && "$1" =~ "python.*" ]]; then
+    PYTHON=$1
+fi
 
 set -eo pipefail
 
@@ -21,14 +27,14 @@ set -eo pipefail
 export PYTHONUNBUFFERED=1
 
 # Install nox
-python3.9 -m pip install --upgrade --quiet nox pip
-python3.9 -m nox --version
+${PYTHON} -m pip install --upgrade --quiet nox pip
+${PYTHON} -m nox --version
 
 # When NOX_SESSION is set, it only runs the specified session
 if [[ -n "${NOX_SESSION:-}" &&  ( "$NOX_SESSION" == "integration_postgres" || "$NOX_SESSION" == "integration_sql_server" || "$NOX_SESSION" == "integration_mysql" || "$NOX_SESSION" =~ integration_oracle.* ) ]]; then
-    ./cloud_sql_proxy -instances="$CLOUD_SQL_CONNECTION" & python3.9 -m nox --error-on-missing-interpreters -s "${NOX_SESSION:-}"
+    ./cloud_sql_proxy -instances="$CLOUD_SQL_CONNECTION" & ${PYTHON} -m nox --error-on-missing-interpreters -s "${NOX_SESSION:-}"
 elif [[ -n "${NOX_SESSION:-}" ]]; then
-    python3.9 -m nox --error-on-missing-interpreters -s "${NOX_SESSION:-}"
+    ${PYTHON} -m nox --error-on-missing-interpreters -s "${NOX_SESSION:-}"
 else
     echo "NOX_SESSION env var not set"
     exit 1

--- a/ci/build.sh
+++ b/ci/build.sh
@@ -17,7 +17,7 @@
 #
 
 PYTHON="python3.9"
-if [[ -n "$1" && "$1" =~ "python.*" ]]; then
+if [[ -n "$1" && "$1" =~ python.* ]]; then
     PYTHON=$1
 fi
 

--- a/cloudbuild.yaml
+++ b/cloudbuild.yaml
@@ -31,7 +31,9 @@ steps:
   waitFor: ['-']
 - id: unit-alpine
   name: 'python:3.10-alpine'
-  args: ['bash', './ci/build.sh', 'python3.10']
+  script: |
+    apk install bash
+    bash ./ci/build.sh python3.10
   env:
   - 'NOX_SESSION=unit_default_python'
   waitFor: ['-']

--- a/cloudbuild.yaml
+++ b/cloudbuild.yaml
@@ -32,7 +32,7 @@ steps:
 - id: unit-alpine
   name: 'python:3.10-alpine'
   script: |
-    apk install bash
+    apk add bash
     bash ./ci/build.sh python3.10
   env:
   - 'NOX_SESSION=unit_default_python'

--- a/cloudbuild.yaml
+++ b/cloudbuild.yaml
@@ -29,6 +29,12 @@ steps:
   env:
   - 'NOX_SESSION=unit'
   waitFor: ['-']
+- id: unit-alpine
+  name: 'python:3.10-alpine'
+  args: ['bash', './ci/build.sh', 'python3.10']
+  env:
+  - 'NOX_SESSION=unit_default_python'
+  waitFor: ['-']
 - id: proxy-install
   name: 'gcr.io/cloud-devrel-public-resources/python-multi'
   args: ['bash', './ci/install_cloud_proxy.sh']

--- a/noxfile.py
+++ b/noxfile.py
@@ -82,6 +82,11 @@ def unit_small(session):
     unit(session)
 
 
+@nox.session(python=DEFAULT_PYTHON_VERSION, venv_backend="venv")
+def unit_default_python(session):
+    unit(session)
+
+
 @nox.session(python=PYTHON_VERSIONS, venv_backend="venv")
 def samples(session):
     """Run the snippets test suite."""

--- a/setup.py
+++ b/setup.py
@@ -39,7 +39,7 @@ dependencies = [
     "pandas",
     "parsy>=2.1",
     "psycopg2-binary>=2.9.9",
-    "pyarrow==14.0.1",  # ibis-framework 7.1.0 depends on pyarrow<15 and >=2
+    "pyarrow",
     "pydata-google-auth>=1.8.2",
     "PyMySQL>=1.1.1",
     "PyYAML>=6.0.2",

